index.js

```diff
@@ -221,3 +221,82 @@ if (crypto) {
     activator
   );
 }
+
+var instrumentPromise = !!global.Promise;
+
+if(instrumentPromise) {
```

Comment:
this block might need a comment, its here to filter out non native promise implementations.  It checks that the promise implementation does not use some asynchronous method we have already wrapped, and that it does not resolve synchronously